roles/openshift_on_openstack_scaleup/tasks/main.yml

```diff
@@ -1,71 +1,122 @@
 
 # Run the Ansible playbook that creates the OpenStack resources.
 - name: Creating the OpenStack resources
-  shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_openstack_dir }}/openshift-cluster/provision.yml -e openshift_repos_enable_testing=true 2>&1 > {{ openstack_scaleup_log }}"
+#  shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_cluster_directory }}/provision.yml -e openshift_repos_enable_testing=true 2>&1 > {{ openstack_scaleup_log }}"
+  shell: >
+    {{ ansible_playbook }} -vvv
+    --user openshift
+    -i inventory
+    -i {{ inventory_py }}
+    {{ openshift_cluster_directory }}/provision.yml 2>&1 >> {{ openstack_scaleup_log }}
+  args:
+    # Use bash to get the posix style redirects.
+    executable: /bin/bash
+
+# Get the new OpenStack VMs by name.
+- name: Getting the new OpenStack VM network information by name
+  shell: "{{ openstack }} server list --name app-node-{{ item }}.scale-ci.example.com --format value -c Networks"
+  with_sequence: "start={{ start_node_number }} end={{ end_node_number }}"
+  register: networks
+
+# Create a list of addresses from the standard output of the last commands.
+- name: Creating a list of addresses from the OpenStack networks output
+  set_fact:
+    addresses: "{{ addresses }} + ['{{ item['stdout'].split(', ')[1] }}' ]"
+  with_items: "{{ networks['results'] }}"
+
+# Write the group name for the new nodes inventory file.
+- name: Writing the group header for the new nodes inventory file
+  copy:
+    content: "[new-nodes]\n"
+    dest: "{{ new_nodes_inventory }}"
+
+# Create the new inventory file
+- name: Adding the addresses to the new nodes inventory file
+  lineinfile:
+    path: "{{ new_nodes_inventory }}"
+    line: "{{ item }} openshift_node_labels=\"{'region': 'primary'}\""
+  with_items: "{{ addresses }}"
+
+# Configure the time servers on all the new nodes.
+- name: Configuring the time servers on all the new nodes
+  shell: >
+    {{ ansible_playbook }} -vvv
+    --user openshift
+    -i {{ new_node_inventory }}
+    configure_time_servers.yml 2>&1 >> {{ ansible_user_dir }}/time_servers.log"
   args:
     # Use bash to get the posix style redirects.
     executable: /bin/bash
 
-## TODO:
-## Set the path for the OpenShift scaleup log file.
-#- name: Creating the OpenStack scaleup log file variable
-#  set_fact:
-#    openshift_scaleup_log: "{{ ansible_user_dir }}/openshift_scaleup.log"
-#
-## Scaling up existing OpenShift deployment.
-#- name: Scale up the OpenShift resources
-#  shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} openshift-ansible/playbooks/openshift-node/scaleup.yml 2>&1 > {{ openshift_scaleup_log }}"
-#  when: 'groups["new_nodes"] | list'
+# Scale up the OpenShift cluster.
+- name: Scaling up the OpenShift resources
+  shell: >
+    {{ ansible_playbook }} -vvv
+    --user openshift
+    -i inventory
```

Comment:
I know this wasn't in the original code too, but the path to the inventory should also be ideally a var, similarly to inventory_py